scripts/unit_test.sh

```diff
@@ -1,4 +1,12 @@
 #!/usr/bin/env bash
 
-go test -v -coverprofile=coverage.out $(go list ./... | grep -v /tests/ | grep -v '/sdk/')
-#go tool cover -func=coverage.out
+set -ex
+
+coverage_dir=$PWD/coverage/ut # should be under the same parent folder as e2e coverage dir
+
+echo "Re-creating unit test coverage directory: $coverage_dir"
+rm -rf $coverage_dir
+mkdir -p $coverage_dir
+
+go test -cover -v  $(go list ./... | grep -v /tests/ | grep -v '/sdk/') -args -test.gocoverdir=$coverage_dir
```

Comment:
I don't believe we need to specify dir. you can just specify -coverprofile=$coveragedir/coverage.out